### PR TITLE
Improve protocol to simplify remoting

### PR DESCRIPTION
**Pull request by int19h**
Change protocol so that the type of every message (one-way notification, request, response) is encoded in the message name in a uniform way.

Rename associated APIs (send_message etc) for clarity.

### CODE REVIEW

**Comment by int19h** [comment]
Corresponding changes in R-Host coming shortly.

**Comment by int19h** [comment]
The purpose of this is to enable the broker to figure out how to handle the packet by only looking at its name (so all custom messages sent from e.g. R code do not need to be special-cased). With the change, there are three types of messages:

Notifications: `!Foo`
Requests: `?Foo`
Responses: `:Foo`

So first character alone determines message type.

Also, I got rid of most special character names for the sake of clarity, keeping them only for those messages that constitute the bulk of the typical session: input prompts, evals, console output, and busy status reporting.

File: src/Host/Client/Impl/Definitions/IRCallbacks.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ public interface IRCallbacks {
         /// </summary>
         /// <param name="url"></param>
         /// <returns></returns>
-        Task Browser(string url);
+        Task WebBrowser(string url);
```
Comment (int19h) [inline src/Host/Client/Impl/Definitions/IRCallbacks.cs]: This is to disambiguate from R debug browser.

File: src/Host/Client/Impl/Host/RHost.cs [hunk 2]
```diff
@@ -131,18 +131,22 @@ public sealed partial class RHost : IDisposable, IRExpressionEvaluator {
...
             return id;
         }
 
         private async Task<string> RespondAsync(Message request, CancellationToken ct, params object[] args) {
+            Debug.Assert(request.Name.StartsWithOrdinal("?"));
             TaskUtilities.AssertIsOnBackgroundThread();
 
             string id;
-            var message = CreateMessage(out id, ":", request.Id, request.Name, args);
+            var message = CreateMessage(out id, ":" + request.Name.Substring(1), request.Id, args);
```
Comment (MikhailArkhipov) [inline src/Host/Client/Impl/Host/RHost.cs]: Throw if Name is actually shorter than 1 and does not start with '?' ?

**Comment by MikhailArkhipov** [comment]
:+1:

**Pull request merged**